Consistent session cookies across haproxies in HA apps

codificat opened the pull request:
Make the local-gear set a cookie value using the gear's DNS name, the same that remote gears do, so that HA applications get consistent session cookie values across multiple haproxies.

See https://bugzilla.redhat.com/1377433

thrasher-redhat commented on cartridges/openshift-origin-cartridge-haproxy/usr/bin/update-cluster:
```diff
@@ -111,8 +111,11 @@ fi
 
 # Add the local gear
+# For cookie consistency with remote gears we'll use the gear's DNS name here,
+# whic takes the form:   gearname-domain
```
Minor spelling in comment - "which"

codificat replied:
Thanks, will fix the `which` and re-push.

Remote gears's cookies are set in line 76: `$gear_name` there comes from the arguments in the `update-cluster` invocation, which are a list of `gear_dns_name|endpoint`; line 49 sets `gear_name` based on the DNS name of the gears, so this change does the same with the local gear.

codificat replied:
@thrasher-redhat comments updated a bit to hopefully make it more clear

thrasher-redhat reviewed (commented):
Looks good as far as I can tell.  Do you know where the remote gear's cookie is set?  Want to verify that its always going to be of that "GEAR_NAME-NAMESPACE" form.

codificat committed:
BZ 1377433: consistent session cookie values

Make the local-gear set a cookie value using the gear's DNS name,
the same that remote gears do, so that HA applications get consistent
session cookie values across multiple haproxies.

Fixes https://bugzilla.redhat.com/1377433

thrasher-redhat commented:
LGTM

Miciah commented:
[test][extended:cartridge,gear]

Miciah commented:
`AWS::EC2::Errors::InternalError` while provisioning. Please [test] again.

thrasher-redhat commented:
@Miciah - was there anything else needed for this PR or are we good to merge?

Miciah commented:
@thrasher-redhat, good to merge.

openshift-bot, please [merge]!

Miciah commented:
openshift-bot, please try to [merge] again!

Miciah commented:
openshift-bot, please really [merge]!

Miciah commented:
Hoping this is a fluke:
```
# cucumber --strict -f progress -f rerun --out /tmp/rerun_ae323edba1d85775675592845950d773.txt  -f junit --out /tmp/rhc/cucumber_results -t @node3 -t ~@not-rhel -t ~@not-origin -r openshift-test/tests @/tmp/rerun_e6125d5278aed18ac2acd2c702da3d6a.txt

Using RSpec 2
simple: 30784 ruby 1.9.3 2013-11-22 [x86_64-linux]
F-----

(::) failed steps (::)

Could not create application 1 (RuntimeError)
./openshift-test/controller/test/cucumber/step_definitions/application_steps.rb:54:in `/^a new client created( scalable)? (.+) application$/'
openshift-test/tests/platform-endpoints.feature:6:in `Given a new client created scalable mock-0.3 application'

Failing Scenarios:
cucumber openshift-test/tests/platform-endpoints.feature:5 # Scenario: SSL to Gear is specified in the cartridge's manifest.yml
```

openshift-bot, please try one more time to [merge]!

Miciah commented:
Yay! Thanks!

Miciah commented:
Yay! Thanks!
